src/sample.rs

```diff
@@ -27,173 +29,285 @@ use std::str::FromStr;
-					}
-					let id = SubscriptionId::new(subscriptionid.unwrap());
-					let kinds_vec: Vec<&str> = kinds_raw.unwrap().split(',').collect();
-					let mut kinds = Vec::with_capacity(kinds_vec.len());
-					for kind in kinds_vec {
-						if let Ok(k) = Kind::from_str(kind) {
-							kinds.push(k);
-						}
-					}
-					let since = Timestamp::from_str(since_raw.unwrap()).unwrap();
-					let until = Timestamp::from_str(until_raw.unwrap()).unwrap();
-					let filter = Filter::new().kinds(kinds).since(since).until(until);
-					let client_message = ClientMessage::new_req(id, vec![filter]);
-					let serialized_message = client_message.as_json();
-					tx.unbounded_send(Message::text(serialized_message)).unwrap();
-				},
-				"closesubscription" => {
-					let subscriptionid = words.next();
-					if subscriptionid.is_none() {
-						println!("ERROR: closesubscription has 1 required argument: `closesubscription subscriptionid");
-					}
-					let id = SubscriptionId::new(subscriptionid.unwrap());
-					let client_message = ClientMessage::close(id);
-					let serialized_message = client_message.as_json();
-					tx.unbounded_send(Message::text(serialized_message)).unwrap();
-				},
-				"shutdown" => {
-					tx.unbounded_send(Message::Close(None)).unwrap();
-					tx.close_channel();
-					println!("Civkit sample exiting...");
-					process::exit(0x0100);
-				},
-				_ => { println!("Unknown command !"); },
-			}
-		}
-	}
+    println!("Civkit sample startup successful. Enter \"help\" to view available commands");
+
+    loop {
+        print!("> ");
+        io::stdout().flush().unwrap();
+        let mut line = String::new();
+        if let Err(e) = io::stdin().read_line(&mut line) {
+            break println!("ERROR {}", e);
+        }
+
+        if line.len() == 0 {
+            continue;
+        }
+
+        match respond(&line, &tx, &client_keys) {
+            Ok(quit) => {
+                if quit {
+                    process::exit(0x0100);
+                }
+            }
+            Err(err) => {
+                write!(std::io::stdout(), "{err}").expect("error: Failed to write to stdout");
+                std::io::stdout()
+                    .flush()
+                    .expect("error: Failed to flush stdout");
+            }
+        }
+    }
+}
+
+fn cli() -> Command {
+    // strip out usage
+    const PARSER_TEMPLATE: &str = "\
+        {all-args}
+    ";
+    // strip out name/version
+    const APPLET_TEMPLATE: &str = "\
+        {about-with-newline}\n\
+        {usage-heading}\n    {usage}\n\
+        \n\
+        {all-args}{after-help}\
+    ";
+
+    Command::new("Nostr Client REPL")
+        .multicall(true)
+        .arg_required_else_help(true)
+        .subcommand_required(true)
+        .help_template(PARSER_TEMPLATE)
+        .subcommand(
+            Command::new("sendtextnote")
+                .args([arg!(-c --content <CONTENT> "Send a text note to the relay")])
```

Comment:
This adds `content` as an option to the `sendtextnote` which results to the need of always prefixing the content with a flag. This pattern doesn't feel intuitive and makes the repl commands more complicate as `content` should be an argument of the command and not an option. Based on the [documentation of `Command`](https://docs.rs/clap/latest/clap/struct.Command.html) the right way to do it is using `Arg::new`: 
```suggestion
                .args([Arg::new("content").help("the content of the text note").required(true)])
```

This change also removes the need for checking if `content.is_none()` when matching because you can define the argument as required.

I believe that this comment applies to all the available commands of the repl; the `arg!(...)` pattern can change to `Arg::new(...).help(...).required(true)` to actually define arguments instead of options and simultaneously remove the need for args existence checking inside the individual commands. You can verify that this is indeed a change of behavior because of the need to add the flags in a74947c5d001b7cdf91fab18a67cd1771fbe2cb5.

Reply:
Makes sense!